packages/mui-base/src/ListboxUnstyled/defaultListboxReducer.ts

```diff
@@ -228,6 +228,74 @@ function handleBlur<TOption>(state: ListboxState<TOption>): ListboxState<TOption
   };
 }
 
+export const textCriteriaMatches = <TOption>(
+  nextFocus: TOption,
+  textCriteriaStr: string,
+  stringifyOption: (option: TOption) => string | undefined,
+) => {
+  const text = stringifyOption(nextFocus)?.trim().toLowerCase();
+
+  if (!text || text.length === 0) {
+    // Make option not navigable if stringification fails or results in empty string.
+    return false;
+  }
+
+  return text.indexOf(textCriteriaStr) === 0;
+};
+
+function handleTextNavigation<TOption>(
+  state: ListboxState<TOption>,
+  textCriteriaStr: string,
+  optionStringifier: (option: TOption) => string | undefined,
```

Comment:
`optionStringifier` is a part of `props`. There's no need to have it as a separate parameter.